python/pyspark/ml/tests/connect/test_parity_torch_distributor.py

```diff
@@ -0,0 +1,183 @@
+        # create temporary directory for Worker resources coordination
+        self.tempdir = tempfile.NamedTemporaryFile(delete=False)
+        os.unlink(self.tempdir.name)
+        os.chmod(
+            self.gpu_discovery_script_file.name,
+            stat.S_IRWXU | stat.S_IXGRP | stat.S_IRGRP | stat.S_IROTH | stat.S_IXOTH,
+        )
+
+        self.spark = (
+            SparkSession.builder.appName(class_name)
+            .config("spark.test.home", SPARK_HOME)
+            .config("spark.driver.resource.gpu.amount", "3")
+            .config(
+                "spark.driver.resource.gpu.discoveryScript", self.gpu_discovery_script_file.name
+            )
+            .remote("local-cluster[2,2,1024]")
+            .getOrCreate()
+        )
+
+        self.mnist_dir_path = tempfile.mkdtemp()
+
+    def tearDown(self) -> None:
+        shutil.rmtree(self.mnist_dir_path)
+        os.unlink(self.gpu_discovery_script_file.name)
+        self.spark.stop()
+
+    # TODO(SPARK-42994): Support sc.resources
+    @unittest.skip("need to support sc.resources")
+    def test_get_num_tasks_locally(self):
+        super().test_get_num_tasks_locally()
+
+    # TODO(SPARK-42994): Support sc.resources
+    @unittest.skip("need to support sc.resources")
+    def test_get_gpus_owned_local(self):
+        super().test_get_gpus_owned_local()
+
+    # TODO(SPARK-42994): Support sc.resources
+    @unittest.skip("need to support sc.resources")
+    def test_local_training_succeeds(self):
+        super().test_local_training_succeeds()
+
+
+@unittest.skipIf(not have_torch, "torch is required")
+class TorchDistributorDistributedUnitTestsOnConnect(
+    TorchDistributorDistributedUnitTestsMixin, unittest.TestCase
+):
+    def setUp(self) -> None:
+        class_name = self.__class__.__name__
+        self.gpu_discovery_script_file = tempfile.NamedTemporaryFile(delete=False)
+        self.gpu_discovery_script_file.write(
+            b'echo {\\"name\\": \\"gpu\\", \\"addresses\\": [\\"0\\",\\"1\\",\\"2\\"]}'
+        )
+        self.gpu_discovery_script_file.close()
+        # create temporary directory for Worker resources coordination
+        self.tempdir = tempfile.NamedTemporaryFile(delete=False)
+        os.unlink(self.tempdir.name)
+        os.chmod(
+            self.gpu_discovery_script_file.name,
+            stat.S_IRWXU | stat.S_IXGRP | stat.S_IRGRP | stat.S_IROTH | stat.S_IXOTH,
+        )
+        self.spark = (
+            SparkSession.builder.appName(class_name)
+            .config("spark.test.home", SPARK_HOME)
+            .config(
+                "spark.worker.resource.gpu.discoveryScript", self.gpu_discovery_script_file.name
+            )
+            .config("spark.worker.resource.gpu.amount", "3")
+            .config("spark.task.cpus", "2")
+            .config("spark.task.resource.gpu.amount", "1")
+            .config("spark.executor.resource.gpu.amount", "1")
+            .remote("local-cluster[2,2,1024]")
```

Comment:
Can we reuse the similar code in spark.ml side ?

Reply:
no, I find in Connect `builder.config` won't accept a `SparkConf`

Reply:
We can still write a helper function for this part, only small difference on its arguments

Reply:
ok, let me add one